libcontainer/init_linux.go

```diff
@@ -499,6 +499,21 @@ func signalAllProcesses(m cgroups.Manager, s os.Signal) error {
 	if err := m.Freeze(configs.Frozen); err != nil {
 		logrus.Warn(err)
 	}
```

Comment:
Again, the whole point of introducing cgroup.kill is you don't have to freeze/thaw, and yet you do that.

Reply:
@kolyshkin ah i moved it after this in latest commit but as per man-pages it says

```
+	Killing a cgroup tree will deal with concurrent forks appropriately and
+	is protected against migrations. If callers require strict guarantees
+	they can issue the cgroup.kill request after a freezing the cgroup via
+	cgroup.freeze.```

Reply:
Please see the discussion between @brauner and @rgushchin, which stars [here]( https://lwn.net/ml/cgroups/YIt32/aQJfkw53ic@carbon.dhcp.thefacebook.com/).

In short, the sentence you quote was only in v1 of the patchset and it was dropped later. IOW freezing is not required, and it's one of the "selling points" of this feature.

Reply:
@kolyshkin thanks for pointing out this discussion this has been fixed in last commit.